### DISCUSSION TITLE
switchboard preparation not working

### DISCUSSION THREAD

**Discussion by RuABraun** [lhotse-speech/lhotse#803] [Q&A] [answered]
Having problems preparing switchboard, hoping someone here can help. At my institute we have these dirs for Switchboard:
```
Switchboard_2_Phase_II/     Switchboard_2_Phase_III/    Switchboard_Cellular_Part1/ Switchboard_Cellular_Part2
```
I get this error when running lhotse prepare with `Switchboard_Cellular_Part1/`
```
  File "/idiap/temp/rbraun/programs/anaconda3/lib/python3.7/site-packages/lhotse/recipes/switchboard.py", line 57, in prepare_switchboard                                                         
    groups.append({'audio': ap, 'text-0': name_to_text[f'{name}A'], 'text-1': name_to_text[f'{name}B']})                                                                                                   
KeyError: 'sw_40000A'
```
I notice that in `Switchboard_Cellular_Part1/dbase/data` I have an sph file `sw_40000.sph`, but I guess because it has no A the key error is occuring?

It's not clear to me whether I have weird versions of the corpus or if lhotse is doing something wrong.

**Comment by pzelasko** [comment, marked as answer]
I think you may have a different switchboard distribution that is not the target of the current recipe. The current recipe is just for phase I, please see description:

https://github.com/lhotse-speech/lhotse/blob/115b564dca7188c2d1f09f59bd5aa78f491e2db9/lhotse/recipes/switchboard.py#L4-L9

I think if you'd like to build and contribute a recipe for the other packages, you might find the existing recipe helpful as a starting point.

**Reply by RuABraun** [reply]
Yeah I saw the help, I thought `Switchboard_Cellular_Part1/2` might correspond to Phase 1 but I guess not.

I will try and contribute a solution! Next week tho

edit: turns out other corpuses dont have transcripts, and I got access to the right corpus!
